proxy/src/proxy.rs

```diff
@@ -265,7 +274,7 @@ pub async fn task_main(
     loop {
         tokio::select! {
             accept_result = listener.accept() => {
-                let (socket, _) = accept_result?;
+                let (socket, peer_addr) = accept_result?;
```

Comment:
we can't use this peer_addr. this points to the load balancer. you must use the `ip` variable below as part of the `WithClientIp` flow

Reply:
What we can do is use `ip.unwrap_or(peer_addr)`